snafu/benchmarks/uperf/uperf.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Wrapper for running the uperf benchmark. See http://uperf.org/ for more information."""
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+import re
+import datetime
+import dataclasses
+import shlex
+from snafu.benchmarks import Benchmark, BenchmarkResult
+from snafu.config import Config, ConfigArgument, FuncAction, check_file, none_or_type
+from snafu.process import sample_process, ProcessSample
+
+
+class ParseRangeAction(FuncAction):
+    """Parses node_range and density_range attributes."""
+
+    def func(self, arg: str) -> List[int]:
+        return [int(x) for x in arg.split("-") if x != ""]
+
+
+@dataclasses.dataclass
```

Comment:
I'm glad to see you're taking advantage of the newer `dataclass` type. Should probably mention the Python version requirements in the docs unless I overlooked it.

Reply:
I can add a note in the docs about it for sure, but I did add the dataclasses backport into our dependencies, meaning our version requirements haven't changed.